frontend/src/hooks/useToast.ts

```diff
@@ -1,41 +1,71 @@
-import React from 'react'
+import React, { useMemo } from 'react'
 import {
   RenderProps,
   useToast as useChakraToast,
-  UseToastOptions,
+  UseToastOptions as ChakraUseToastOptions,
 } from '@chakra-ui/react'
 
-import { Toast, ToastProps } from '~/components/Toast/Toast'
+import { Toast, ToastProps, ToastStatus } from '~/components/Toast/Toast'
 
 type ToastBehaviourProps = Pick<
-  UseToastOptions,
+  ChakraUseToastOptions,
   'duration' | 'position' | 'render'
 >
 
 // onClose is provided by the chakra hook and should not be exposed to clients
-export type UseToastProps = Omit<ToastBehaviourProps & ToastProps, 'onClose'>
+export type UseToastProps = Omit<
+  ToastBehaviourProps & ToastProps,
+  'onClose' | 'status'
+> & { status?: ToastStatus }
 
-type UseToast = (
-  // NOTE: Chakra's toast status is different from ours, hence we need to omit it for type compatibility
```

Comment:
should we retain this comment for `UseToastProps`?

Reply:
resolved in 92281db